apps/wpcom-block-editor/src/wpcom/features/tracking/track-record-event.js

```diff
@@ -16,10 +21,21 @@ if ( typeof window !== 'undefined' ) {
 // This means we don't have any extra props like user
 
 export default ( eventName, eventProperties ) => {
-	// Required by Tracks when added manually
-	const blog_id = window._currentSiteId;
-	const site_type = window._currentSiteType;
-	const user_locale = window._currentUserLocale;
+	/*
+	 * Custom Properties.
+	 * Required by Tracks when added manually.
+	 */
+	const customProperties = {
+		blog_id: window._currentSiteId,
+		site_type: window._currentSiteType,
+		user_locale: window._currentUserLocale,
+	};
+
+	// Custom Properties: Populate with selected_block if it's the case.
+	const selectedBlock = select( 'core/block-editor' ).getSelectedBlock();
+	if ( selectedBlock ) {
+		customProperties.selected_block = selectedBlock.name;
+	}
```

Comment:
I don't think we need this for most of our events. Events like deleting blocks, moving them around, undo, redo… most of them already come with a block name or they are block-agnostic. 

I think other changes in this file are good to keep but I'd vote to move these few lines handling the selected block specifically into the search tracking.